rust/arrow/src/array.rs

```diff
@@ -1659,13 +1710,80 @@ mod tests {
         field_types.push(Field::new("a", DataType::Boolean, false));
         field_types.push(Field::new("b", DataType::Int64, false));
         let struct_array_data = ArrayData::builder(DataType::Struct(field_types))
+            .len(4)
+            .add_child_data(boolean_data.clone())
+            .add_child_data(int_data.clone())
+            .build();
+        let struct_array = StructArray::from(struct_array_data);
+
+        assert_eq!(boolean_data, struct_array.column(0).data());
+        assert_eq!(int_data, struct_array.column(1).data());
+    }
+
+    #[test]
+    fn test_struct_array_slice() {
+        let mut null_bits: [u8; 1] = [0; 1];
+        bit_util::set_bit(&mut null_bits, 0);
+        bit_util::set_bit(&mut null_bits, 2);
+        bit_util::set_bit(&mut null_bits, 4);
+        let null_bit_buffer = Buffer::from(null_bits);
```

Comment:
I see that you reuse this bit_buffer on the boolean and int where they're not null. Should I disregard that, as the intention of the test is to show that the struct's offsets are considered correctly in the case of a slice?

Reply:
Sure. I'll use different bit buffer for boolean and int.